### PR TITLE
feat: Add Generate Endpoint Support

### PR DESCRIPTION
**Pull request by nnshah1**
Adds support for endpoint-type 'generate' (https://docs.nvidia.com/deeplearning/triton-inference-server/user-guide/docs/protocol/extension_generate.html).

Adds new endpoint types:

1) generate
2) kserve
3) tritonllm_engine

### CODE REVIEW

File: genai-perf/genai_perf/parser.py [hunk 1]
```diff
@@ -135,47 +162,38 @@ def _check_conditional_args(
...
-                args.output_format = ic.OutputFormat.NVCLIP
-
-            if args.endpoint is not None:
-                args.endpoint = args.endpoint.lstrip(" /")
-            else:
-                args.endpoint = _endpoint_type_map[args.endpoint_type]
-    elif args.endpoint_type is not None:
+    if args.endpoint_type not in _endpoint_type_map:
+        parser.error(f"Invalid endpoint type {args.endpoint_type}")
+        return
```
Comment (nv-hwoo) [inline genai-perf/genai_perf/parser.py:167]: Do we need to return? It looks like [ArgumentParser.error](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.error) terminates the program.

File: genai-perf/genai_perf/parser.py [hunk 2]
```diff
@@ -135,47 +162,38 @@ def _check_conditional_args(
...
+    args.output_format = endpoint_config.output_format
+    if (
+        args.service_kind is not None
+        and args.service_kind != endpoint_config.service_kind
+    ):
         parser.error(
-            "The --endpoint-type option should only be used when using the 'openai' service-kind."
+            f"Invalid service-kind {args.service_kind} for endpoint-type {args.endpoint_type}. service-kind is now inferred from endpoint-type and the argument will be removed in a future release."
         )
+        return
```
Comment (nv-hwoo) [inline genai-perf/genai_perf/parser.py:178]: same as above

File: genai-perf/genai_perf/parser.py [hunk 3]
```diff
@@ -63,14 +64,40 @@ def to_lowercase(self):
...
+    ),
+    "rankings": EndpointConfig("v1/ranking", "openai", ic.OutputFormat.RANKINGS),
+    "vision": EndpointConfig(
+        "v1/chat/completions", "openai", ic.OutputFormat.OPENAI_VISION
+    ),
+    "image_retrieval": EndpointConfig(
+        "v1/infer", "openai", ic.OutputFormat.IMAGE_RETRIEVAL
+    ),
+    "generate": EndpointConfig(
+        "v2/models/{MODEL_NAME}/generate", "openai", ic.OutputFormat.TRITON_GENERATE
```
[the remark this answers is not in the document]
Comment (nnshah1) [reply genai-perf/genai_perf/parser.py:93]: that's correct - we can have a todo item to rename it more generic - its something like a llm / sse client now as supports more than openai - but left it alone in this PR

File: genai-perf/genai_perf/parser.py (hunk 1 above)
Comment (nnshah1) [reply genai-perf/genai_perf/parser.py:167]: that is correct - but I followed the pattern in the file - 

Would want to keep it consistent - but generally if you use parser.error you don't need a return - but maybe its clearer this way?

File: genai-perf/genai_perf/parser.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (nv-hwoo) [reply genai-perf/genai_perf/parser.py:93]: Thanks for clarifying. Yeah, I wouldn't want to change too many things in this PR. We can keep it as is.

**Review by nv-hwoo — DISMISSED** [review]

File: genai-perf/tests/test_cli.py [hunk 4]
```diff
@@ -106,34 +102,30 @@ def test_help_version_arguments_output_and_exit(
...
                 {"endpoint": "v1/chat/completions"},
             ),
             (
-                ["--endpoint-type", "rankings", "--service-kind", "openai"],
+                ["--endpoint-type", "rankings"],
                 {"endpoint": "v1/ranking"},
             ),
             (
-                ["--endpoint-type", "image_retrieval", "--service-kind", "openai"],
+                ["--endpoint-type", "image_retrieval"],
```
Comment (debermudez) [inline genai-perf/tests/test_cli.py:122]: Need to add a test for nvclip here.

File: genai-perf/genai_perf/profile_data_parser/profile_data_parser.py [hunk 5]
```diff
@@ -63,6 +64,7 @@ def __init__(
 
     def _get_profile_metadata(self, data: dict) -> None:
         self._service_kind = data["service_kind"]
+        self._endpoint = data["endpoint"]
```
Comment (nv-hwoo) [inline genai-perf/genai_perf/profile_data_parser/profile_data_parser.py:67]: This variable is never used anywhere. Can we delete it?

**Review by debermudez — COMMENTED** [review]
LGTM. waiting on @nv-hwoo or @dyastremsky to take a peek

**Review by nv-hwoo — APPROVED** [review]

**Pull request merged**